api/sendfild/service/service_test.go

```diff
@@ -0,0 +1,646 @@
+package service
+
+import (
+	"context"
+	"math/rand"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/api/apistruct"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log/v2"
+	mh "github.com/multiformats/go-multihash"
+	"github.com/stretchr/testify/require"
+	"github.com/textileio/go-ds-mongo/test"
+	"github.com/textileio/powergate/v2/lotus"
+	"github.com/textileio/powergate/v2/tests"
+	powutil "github.com/textileio/powergate/v2/util"
+	pb "github.com/textileio/textile/v2/api/sendfild/pb"
+	"github.com/textileio/textile/v2/util"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+const (
+	bufSize = 1024 * 1024
+	oneFil  = 1000000000
+)
+
+var (
+	ctx, _ = context.WithTimeout(context.Background(), 2*time.Minute)
+)
+
+func TestMain(m *testing.M) {
+	powutil.AvgBlockTime = time.Millisecond * 100
+	logging.SetAllLoggers(logging.LevelError)
+
+	cleanup := func() {}
+	if os.Getenv("SKIP_SERVICES") != "true" {
+		cleanup = test.StartMongoDB()
+	}
+	exitVal := m.Run()
+	cleanup()
+	os.Exit(exitVal)
+}
+
+func TestRestartWaiting(t *testing.T) {
```

Comment:
This is a very important one. It tests that the waiting state is properly initialized upon service restart.